File: 10_Fri/Cardio.java
class Cardio{
    public static void main(String[] args) {
        String magazine = "Hello world";
        String message = "HHe old";
        boolean answer = solution1(magazine, message);
        System.out.println(answer);
    }

    public static boolean solution1(String magazine, String message){
        char[] messLetters = message.toCharArray();

        for (char letter : messLetters){
            if (magazine.contains(String.valueOf(letter))|| letter == ' '){
                magazine = magazine.replaceFirst(String.valueOf(letter), "");
                continue;
            }
            else{
                return false;
            }
        }
        return true;
    }

    public static boolean solution2(String magazine, String message){
        char[] messLetters = message.toCharArray();
        int i = 0;

        while (i < messLetters.length){
            if (magazine.contains(String.valueOf(messLetters[i]))|| messLetters[i] == ' '){
                magazine = magazine.replaceFirst(String.valueOf(messLetters[i]), "");
                continue;
            }
            else{
                return false;
            }
        }
        return true;
    }
}

File: 10_Fri/cardio.py
def main():
    magazine = "Hello World"
    message = "He old"
    print(solution2(magazine, message))

def solution1(magazine, message):
    magazine_letters = list(magazine)
    message_letters = list(message)
    for letter in message_letters:
        if letter in magazine_letters and letter != " ":
            magazine_letters.remove(letter)
            continue
        elif letter == " ":
            continue
        else:
            return False 
    return True
        
def solution2(magazine, message):
    magazine_letters = list(magazine)
    message_letters = list(message)
    i = 0
    while i < len(message_letters):
        if message_letters[i] in magazine_letters and message_letters[i] != " ":
            magazine_letters.remove(message_letters[i])
            i+=1
            continue
        elif message_letters[i] == " ":
            i+=1
            continue
        else:
            return False 
        
    return True
    
        

if __name__ == "__main__":
    main();
